### DISCUSSION TITLE
Anyone interested in helping tweak skopt's performance in a benchmarking exercise?

### DISCUSSION THREAD

**Discussion by microprediction** [scikit-optimize/scikit-optimize#1002] [General]
Hello,

  I am wondering if someone with more experience than me is interested in helping tweak skopt's performance in a benchmarking exercise I am running (in the open) for my company. 

  As you can see from the [leaderboard](https://github.com/microprediction/optimizer-elo-ratings/tree/main/results/leaderboards/overall) skopt is beating a whole slew of the usual suspects. Nice job! 

  However, I'd like to include a few more sub-choices in gp_minimize and wondering if anyone has favourite choices of parameters. My use of skopt is [here](https://github.com/microprediction/humpday/blob/main/humpday/optimizers/skoptcube.py). I'm thus far trying the following variations: 

  'lcbexplore':{"acq_func": "LCB", "kappa":3.0},
                             'lcb':{"acq_func":"LCB","kappa":1.96},
                             'lcbexploit':{"acq_func":"LCB","kappa":1.0},
                             'pi':{"acq_func":"pi"},
                             'xi02':{"xi":0.02},
                             'default':{},
                             'sampling':{'acq_optimizer':'sampling'},
                             'lbfgs':{'acq_optimizer':'lbfgs'},
                             'noisy':{'noise':'gaussian'},
                             'calm':{'noise':1e-8},
                             '':{}

  Pls let me know if something else should be tried. 

Peter

**Comment by kernc** [comment]
The leaderboard link 404s.

Wouldn't better-performing combinations of parameters depend always solely on the distribution of the underlying data [that is being fitted](https://github.com/microprediction/humpday/blob/main/humpday/objectives/classic.py) (i.e. [_no free lunch_](https://en.wikipedia.org/wiki/No_free_lunch_in_search_and_optimization))?

Why not just do _grid search_ over _all_ accepted parameters/values?

**Comment by microprediction** [comment]
Thanks, I fixed the leaderboard URL. If you just want one of them, https://github.com/microprediction/optimizer-elo-ratings/tree/main/results/leaderboards/overall

Grid search could be included ... which would probably answer your question. Although remember that these optimizers are being challenged in quite high dimensions sometimes, so it isn't entirely clear what constitutes a grid search.

**Reply by kernc** [reply]
I meant grid search over viable parameters to `gp_minimize()`. (Or even a more informed, skopt-based search (over viable parameters to `gp_minimize()`) — go full meta. :sweat_smile:)

**Reply by microprediction** [reply]
I'm with you now. Just a newbie so not sure how to describe the valid params.

**Reply by kernc** [reply]
I guess just from [the documentation](https://scikit-optimize.github.io/stable/modules/generated/skopt.gp_minimize.html#skopt.gp_minimize)? E.g. `kappa=` is a float parameter, so you can "grid" it as `np.linspace(0, 4, 9)` or "skopt-space" it as `Real(0, 4)` ...
